xicam/core/data/__init__.py

```diff
@@ -1,62 +1,53 @@
 from pathlib import Path
-from typing import Union, List
+from typing import Union, List, Set
 
 from qtpy.QtCore import Signal, QObject
 
 from databroker.utils import ALL
 from databroker.in_memory import BlueskyInMemoryCatalog
-from databroker.core import Header
 from warnings import warn
 
-
-# extension_map = {EDFPlugin: ['edf']}
-
-
 import mimetypes
 import warnings
 
 import entrypoints
 
 
-def ingest_file(filename):
-    """
-    Take in a file name; return a generator that yields (name, doc) pairs.
-    """
-    mimetype = detect_mimetype(filename)
-    ingestor = choose_ingestor(filename, mimetype)
-    generator_instance = ingestor(filename)
-    return generator_instance
-
-
-def detect_mimetype(filename):
+def detect_mimetypes(filename: str) -> List[str]:
     """
     Take in a filename; return a mimetype string like 'image/tiff'.
     """
     # First rely on custom "sniffers" that can employ file signatures (magic
     # numbers) or any other format-specific tricks to extract a mimetype.
+    matched_mimetypes = list()
+
     if Path(filename).is_dir():
-        return None  # TODO: do directories have mime-type?
+        return matched_mimetypes  # TODO: do directories have mime-type?
```

Comment:
It looks like IANA used to have a `text/directory` MIME type, but it has been deprecated (https://www.iana.org/assignments/media-types/media-types.xhtml#text). 

---

Regarding what `detect_mimetypes` should return, it looks like it can return:
* an empty list (which indicates that a directory was passed as `filename`)
* a list of matched mimetypes (no uncaught exceptions, at least one match)

And throws an `UnknownFileType` exception if the MIME type could not be identified (which happens if the file is *not* a directory and no mimetype is discoverable for the file).

Does it make sense to return an empty list to indicate a directory was passed? Should a (at least DEBUG) message be logged?

Reply:
Added higher-level sanity checks